woocommerce/woocommerce-ios
# Product description AI announcement modal: handle CTA to show product creation form

Part of #10021 


## Why


We want to increase the visibility of the product description AI feature, and are showing a modal that has the same design as JITM NMncL2ViLjomfLhuyEhVM1-fi-236_84398. Originally, we wanted to reuse one of the two existing modals - JITM and What's New announcement modal. But each of the remote modal systems has limitations, and we decided to implement a modal locally pecCkj-Cr-p2#comment-382 after syncing JITM and no modal is being shown.

## How

I struggled a bit with how to coordinate the CTA tap event handling to show the product creation form while the modal is dismissed separately through the pre-existing `ModalOverlay`'s `onTapGesture` callback. Ideally, we can await the modal dismissal and then show any UI for the announcement. In the end, the `show` navigation style (pushing a view controller) was fine since we skip the bottom sheet presentation which can result in a UIKit warning when the `dismiss` and `present` are happening the same time.

Also, the modal can be shown when selecting any other main tab since the JITM is synced async. I had to use `AppDelegate.shared.tabBarController?.selectedViewController` to get a navigation controller to show the product creation form. Please share any issues or improvements that we can make here! I think moving the whole modal presentation logic from `DashboardViewController` to `AppCoordinator` would help a lot, but it requires lots of JITM changes.

## Testing instructions


For easier testing, you can update `LocalAnnouncementsProvider.isVisible` to always return true so that you can see the modal consistently.

- Launch the app in the logged-in state and connected to a WPCOM store --> a modal should be shown about the product description AI after seeing `📣 Announcements synced, but nothing received.`/` 📣 There are no announcements to show!` logs in the console
- Tap `Try it now` --> the product creation form should be shown with the physical product type

## Screenshots



https://github.com/woocommerce/woocommerce-ios/assets/1945542/39abd2d9-fe47-49c1-9ed8-401926a3a2d9



---
